web2/hooks/queries/mapper.ts

```diff
@@ -0,0 +1,146 @@
+import {
+  INftCollection,
+  INftItem,
+  IGivingsOffer,
+  IGivingsItem,
+  ITrade,
+  IReceivingsOffer,
+  IReceivingsItem,
+} from "@/interfaces";
+import {
+  Collection,
+  CollectionItem,
+  GivingsOffer,
+  GivingsOfferItem,
+  ReceivingsOffer,
+  ReceivingsOfferItem,
+  Trade,
+} from "@/packages/graphql/generated";
+import {
+  IterableElement,
+  PartialDeep,
+  Primitive,
+  ReadonlyDeep,
+} from "type-fest";
+import { ConditionalSimplifyDeep as SimplifyDeep } from "type-fest/source/conditional-simplify";
+
+type ToMapped<TFull, TPartial extends PartialDeep<TFull>, TMapped> = {
+  [Key in Extract<keyof TMapped, keyof TPartial>]:
+  TMapped[Key] extends Primitive
+    ? TMapped[Key]
+    : TMapped[Key] extends any[] | null
+  // @ts-ignore
+  ? ToMapped<IterableElement<TFull[Key]>, IterableElement<TPartial[Key]>, IterableElement<TMapped[Key]>>[] | null
+  : TMapped[Key] extends any[] ?
+  // @ts-ignore
+      ToMapped<IterableElement<TFull[Key]>, IterableElement<TPartial[Key]>, IterableElement<TMapped[Key]>>[]
+  : TMapped[Key] extends any | null ?
+  // @ts-ignore
+      ToMapped<TFull[Key], TPartial[Key], TMapped[Key]> | null
+  // @ts-ignore
+  : ToMapped<TFull[Key], TPartial[Key], TMapped[Key]>;
+};
```

Comment:
This should now work as expected but please double check it

```suggestion
type ToMappedNullable<TFull, TPartial, TMapped> =
  TMapped extends any[]
    ? ToMapped<IterableElement<TFull>, IterableElement<TPartial>, IterableElement<TMapped>>[]
    : ToMapped<TFull, TPartial, TMapped>;

type ToMapped<TFull, TPartial, TMapped> = {
  [Key in Extract<keyof TMapped, keyof TPartial>]:
    TMapped[Key] extends Primitive
      ? TMapped[Key]
      : null extends TMapped[Key]
      // @ts-ignore
      ? ToMappedNullable<TFull[Key], TPartial[Key], NonNullable<TMapped[Key]>> | null
      : TMapped[Key] extends any[]
      // @ts-ignore
      ? ToMapped<IterableElement<TFull[Key]>, IterableElement<TPartial[Key]>, IterableElement<TMapped[Key]>>[]
      // @ts-ignore
      : ToMapped<TFull[Key], TPartial[Key], TMapped[Key]>;
};
```